serverless/serverless
# 2116 consistent errors missing config

## What did you implement:

Closes #2116

When running a serverless command which reads serverless config during execution in a path without an existing serverless config file, the messaging presented to the dev was inconsistent. It was picked up and reported by two plugins, but the remainder failed to identify the issue and either failed silently or reported that the command was not found.

## How did you implement it:

Validation of the command being run in a serverless directory was done on a per-command basis. This was changed so that the validation is performed within the plugin manager during execution, catching a flag defined in the plugin configuration.

## How can we verify it:

`git clone https://github.com/shanehandley/serverless`
`git co 2116-consistent-errors-missing-config`

Then run the command via `bin/serverless` example:

`bin/serverless config` should correctly fail with `This command can only be run in a Serverless service directory`

This should report correctly for applicable commands:

| Command|
| ------------- |
|Info|
|deploy|
|config|
|install|
|invoke|
|logs|
|metrics|
|package|
|print|
|remove|
|rollback|
|slstats|



## Documentation

No documentation has been added at this stage, as it provides a non-essential developer convenience to built-in commands. 

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
